dpdk/vr_dpdk_virtio.c

```diff
@@ -60,6 +60,35 @@ vr_dpdk_virtio_ntxqs(struct vr_interface *vif)
 }
 
 /*
+ * dpdk_virtio_rx_queue_release - releases a virtio RX queue.
+ *
+ * Returns nothing.
+ */
+static void
+dpdk_virtio_rx_queue_release(unsigned lcore_id, struct vr_interface *vif)
+{
+    struct vr_dpdk_lcore *lcore = vr_dpdk.lcores[lcore_id];
+    struct vr_dpdk_queue *rx_queue = &lcore->lcore_rx_queues[vif->vif_idx];
+    struct vr_dpdk_queue_params *rx_queue_params
+                        = &lcore->lcore_rx_queue_params[vif->vif_idx];
+
+    /* remove the ring from the list of rings to push */
+    dpdk_ring_to_push_remove(rx_queue_params->qp_ring.host_lcore_id,
```

Comment:
I needed to remove line:
if (rx_queue->rxq_ops.f_free(rx_queue->q_queue_h)) {
and it's equivalent for tx queue. Calling .f_free() caused segfault, because there are no .f_free and .f_create for virtio. Now removing virtio interfaces seem to work.